### PR TITLE
Bluetooth: controller: Handle peripheral CIS creation latency

### PR DESCRIPTION
**Pull request by mtpr-ot**
Instead of asserting, handle CIS creation latency for the peripheral by finding next valid ISO event at which to start the CIG ticker.

Refactored calculations of the cig_offset to separate functions, to be used also when calculating a new start offset.

### CODE REVIEW

**Comment by mtpr-ot** [comment]
CI error seems unrelated

**Comment by mtpr-ot** [comment]
Rebased

File: subsys/bluetooth/controller/ll_sw/ull_conn_iso.c [hunk 1]
```diff
@@ -913,22 +917,66 @@ void ull_conn_iso_start(struct ll_conn *conn, uint16_t cis_handle,
...
+			 */
+			uint32_t lost_cig_events;
+			uint32_t iso_interval_us;
+			uint32_t acl_latency_us;
+			uint32_t lost_payloads;
+			uint32_t cis_offset;
+
+			acl_latency_us = instant_latency * conn->lll.interval * CONN_INT_UNIT_US;
+			iso_interval_us = cig->iso_interval * ISO_INT_UNIT_US;
+			lost_cig_events = 1U;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_conn_iso.c:942]: Moved into `else` clause

```suggestion
```

File: subsys/bluetooth/controller/ll_sw/ull_conn_iso.c [hunk 2]
```diff
@@ -913,22 +917,66 @@ void ull_conn_iso_start(struct ll_conn *conn, uint16_t cis_handle,
...
+				 */
+				lost_cig_events = DIV_ROUND_UP(acl_latency_us - cis->offset,
+							       iso_interval_us);
+				cis_offset = cis->offset + (lost_cig_events * iso_interval_us) -
+					     acl_latency_us;
+			} else {
+				/* Latency is less than- or equal to one ISO interval - start at
+				 * next ISO event.
+				 */
+				cis_offset = cis->offset + iso_interval_us - acl_latency_us;
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_conn_iso.c:957]:
```suggestion
				lost_cig_events = 1U;
				cis_offset = cis->offset + iso_interval_us - acl_latency_us;
```

**Review by cvinayak — CHANGES_REQUESTED** [review]
Minor change request.

**Comment by mtpr-ot** [comment]
Sorry, forgot about this; Will update ASAP.

**Review by cvinayak — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**